gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -218,7 +219,8 @@ private boolean isPathWhiteListed(String path) {
         || path.startsWith(V1_INFO_PATH)
         || path.startsWith(V1_NODE_PATH)
         || path.startsWith(UI_API_STATS_PATH)
-        || path.startsWith(OAUTH_PATH);
+        || path.startsWith(OAUTH_PATH)
+        || extraWhitelistPaths.stream().anyMatch(s -> path.startsWith(s));
```

Comment:
To make this code look cleaner, I am wondering if all the paths could be derived from a config list.
We can provide a default (hardcoded initialization) list for the existing paths and then append the `extraWhitelistPaths`
Also, how much overhead is `stream` going to add? This should be a really performance code.

Reply:
Have not looked at the code but... will this only load at start up time of the server.. then performance wont matter.. also wont matter if the reload is done on demand via a trigger in the UI or a REST endpoint .. I would prefer these two methods over reloading automatically all the time. That would only be okay if we know there is basically no performance impact.

Reply:
This code get's called for every request sent to the cluster and hence the comment.

Reply:
Java should evaluate chained `||` operators lazily, so as soon as any of these conditions are true it will skip the rest. Because of that, I think it is a good idea to retain the current structure and order them so that the most latency sensitive paths such as `v1/statement` are first in the list, even if it is a little ugly. 

> Are we expecting any paths to be whitelisted in near future?
This is useful for commercial extensions to Trino such as Starburst. Starburst has extra endpoints for its features such as data products and the access control API.

Lastly, we could clean up this list. At least `/ui/api/stats` should probably be removed. If someone is using it (it is an internal endpoint, not intended to be exposed but who knows...) they could add it back in through the new config.

Reply:
We can do that cleanup later